PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/StorageUploadWithoutLengthCheck.java

```diff
@@ -0,0 +1,108 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.*;
+import org.json.*;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * This class extends the LocalInspectionTool and is used to inspect the usage of Azure Storage upload APIs in the code.
+ * It checks if the upload methods are being called without a 'length' parameter of type 'long'.
+ * The methods to check are defined in a JSON configuration file.
+ * If such a method call is detected, a problem is registered with the ProblemsHolder.
+ */
+
+public class StorageUploadWithoutLengthCheck extends LocalInspectionTool {
+    @Nls
+    @NotNull
+    @Override
+    public String getDisplayName() {
+        return "Use Storage APIs With Length Parameter";
+    }
+
+    private static final List<String> METHODS_TO_CHECK = getMethodsToCheck();
+
+    private static List<String> getMethodsToCheck() {
+        try {
+            InputStream inputStream = StorageUploadWithoutLengthCheck.class.getClassLoader().getResourceAsStream("META-INF/ruleConfigs.json");
+            if (inputStream == null) {
+                throw new FileNotFoundException();
```

Comment:
```suggestion
                throw new FileNotFoundException("Configuration file not found");
```
Error messages should be more informative.